### PR TITLE
remove special case for the no-op first statement in `fully_eliminated`

### PR DESCRIPTION
**Pull request by aviatesk**
It looks like the special case is no longer needed. If this gets broken by future changes, I would like fix it or mark the test cases as `broken`.

### CODE REVIEW

File: test/compiler/irutils.jl [hunk 1]
```diff
@@ -42,24 +42,15 @@ fully_eliminated(@nospecialize args...; retval=(@__FILE__), kwargs...) =
...
-        return isreturn(code[end])
+    length(code) == 1 || return false
+    retstmt = only(code)
+    isreturn(retstmt) || return false
+    retval === (@__FILE__) && return true
+    retval′ = retstmt.val
+    if retval′ isa QuoteNode
+        val = retval′.value
     end
+    return val == retval
```
Comment (aviatesk) [inline test/compiler/irutils.jl:53]:
```suggestion
    return retval′ == retval
```

File: test/compiler/irutils.jl [hunk 2]
```diff
@@ -42,24 +42,15 @@ fully_eliminated(@nospecialize args...; retval=(@__FILE__), kwargs...) =
...
-            code[i] === nothing || return false
-        end
-        return isreturn(code[end])
+    length(code) == 1 || return false
+    retstmt = only(code)
+    isreturn(retstmt) || return false
+    retval === (@__FILE__) && return true
+    retval′ = retstmt.val
+    if retval′ isa QuoteNode
+        val = retval′.value
```
Comment (aviatesk) [inline test/compiler/irutils.jl:51]:
```suggestion
        retval′ = retval′.value
```

**Comment by vtjnash** [comment]
It looks like maybe this was to make the test run correctly with `--code-coverage`?

**Comment by aviatesk** [comment]
I don't think so because this tweak was introduced since a change on the CFG transformation  introduced spurious no-op statement (, which seems to have been fixed lately, xref: #50924), and AFAIU our `compiler` test suite isn't supposed to be run under `--code-coverage` enabled.

**Comment by vtjnash** [comment]
It does run, though many of these tests fail https://buildkite.com/julialang/julia-master-scheduled/builds/485#018a6eac-3dcc-48c3-bd92-915204cd3903

**Comment by aviatesk** [comment]
When we are going to fix the failures in coverage test runt, I'd like to add an explicit check of `if coverage_enabled(...)` within `fully_eliminated`.

**Pull request merged**